ledger-core/virtual/callregistry/objects_request_table.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package callregistry
+
+import (
+	"github.com/insolar/assured-ledger/ledger-core/insolar/payload"
+	"github.com/insolar/assured-ledger/ledger-core/reference"
+	"github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
+)
+
+// ObjectsResultCallRegistry is used for store results for all processed requests per object by object reference.
+type ObjectsResultCallRegistry struct {
+	objects map[reference.Global]ObjectCallResults
+}
+
+type ObjectCallResults struct {
+	CallResults map[reference.Global]CallSummary
+}
+
+func NewObjectRequestTable() ObjectsResultCallRegistry {
+	return ObjectsResultCallRegistry{
+		objects: make(map[reference.Global]ObjectCallResults),
+	}
+}
+
+func (ort ObjectsResultCallRegistry) GetObjectCallResults(objectRef reference.Global) (ObjectCallResults, bool) {
+	workingTable, ok := ort.objects[objectRef]
+	return workingTable, ok
+}
+
+func (ort ObjectsResultCallRegistry) AddObjectCallResults(objectRef reference.Global, callResults ObjectCallResults) bool {
+	if _, exist := ort.objects[objectRef]; exist {
+		return false
+	}
+	ort.objects[objectRef] = callResults
+
+	return true
+}
+
+func (ort ObjectsResultCallRegistry) AddObjectCallResult(objectRef reference.Global, reqRef reference.Global, result *payload.VCallResult) {
+	requests, ok := ort.GetObjectCallResults(objectRef)
+	if !ok {
+		// we should have summary result for object if we finish operation.
+		panic(throw.Impossible())
+	}
+
+	callSummary, ok := requests.CallResults[reqRef]
+	if !ok {
+		// if we not have summary in resultMap, it mean we do not go through stepStartRequestProcessing,
+		// and we can't have a result here
+		panic(throw.Impossible())
+	}
+
+	if callSummary.Result != nil {
+		// we should not have result because migration was before we publish result in object
+		panic(throw.Impossible())
```

Comment:
throw.InvalidState() and all above (it is possible, but something went wrong)

Reply:
done